getline.go

```diff
@@ -33,8 +33,9 @@ var _ tea.Model = (*Editor)(nil)
 func (m *Editor) Update(imsg tea.Msg) (tea.Model, tea.Cmd) {
 	switch msg := imsg.(type) {
 	case tea.WindowSizeMsg:
-		m.Model.SetSize(msg.Width, msg.Height)
-
+		if msg.Width != 0 || msg.Height != 0 {
```

Comment:
Are you sure that's what you want? What if one is zero and the other is not zero?

I think you might want `msg.Width != 0 && msg.Height != 0` instead.

Also, can you add unit tests for this thanks.

Reply:
Done. I tried adding a test but it looks like the whole test is failing?

```
~/bubbline/editline$ go test
go: downloading github.com/rogpeppe/go-internal v1.6.1
--- FAIL: TestBubbline (0.28s)
    --- FAIL: TestBubbline/auto_enter (0.10s)
        driver.go:474: testdata/auto_enter:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/auto_enter:1:
             reset
            resize 40 25
            expected:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/autocomplete_choice (0.00s)
        driver.go:474: testdata/autocomplete_choice:1: applying command "reset" via model updater
        driver.go:474: testdata/autocomplete_choice:1: applying command "set_autocomplete_2" via model updater
        getline_test.go:32: 
            testdata/autocomplete_choice:1:
             reset
            resize 40 10
            set_autocomplete_2
            expected:
            TEA WINDOW SIZE: {40 10}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 10}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/autocomplete_simple (0.00s)
        driver.go:474: testdata/autocomplete_simple:1: applying command "reset" via model updater
        driver.go:474: testdata/autocomplete_simple:1: applying command "set_autocomplete_1" via model updater
        getline_test.go:32: 
            testdata/autocomplete_simple:1:
             reset
            resize 40 25
            set_autocomplete_1
            expected:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/bad_resize (0.00s)
        driver.go:474: testdata/bad_resize:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/bad_resize:1:
             reset
            resize 40 25
            resize 0 0
            output didn't match expected:
            @@ -1,6 +1,6 @@
             TEA WINDOW SIZE: {40 25}
             TEA WINDOW SIZE: {0 0}
             -- view:
            ->                                     ␤
            -M-? toggle key help • C-d erase/stop🛇
            +>                                     ␤
            +M-? toggle key help • C-d erase/stop🛇
             
    --- FAIL: TestBubbline/debug (0.00s)
        driver.go:474: testdata/debug:1: applying command "reset" via model updater
        driver.go:474: testdata/debug:1: applying command "enable_debug" via model updater
        getline_test.go:32: 
            testdata/debug:1:
             reset
            resize 80 25
            enable_debug
            expected:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            
            found:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
    --- FAIL: TestBubbline/end_of_input (0.00s)
        driver.go:474: testdata/end_of_input:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/end_of_input:1:
             reset
            resize 40 25
            expected:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/external_edit (0.00s)
        driver.go:474: testdata/external_edit:1: applying command "reset" via model updater
        driver.go:474: testdata/external_edit:1: applying command "set_editor_env" via model updater
        getline_test.go:32: 
            testdata/external_edit:1:
             reset
            resize 80 25
            set_editor_env
            expected:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            
            found:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
    --- FAIL: TestBubbline/focus_blur (0.02s)
        driver.go:474: testdata/focus_blur:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/focus_blur:1:
             reset
            resize 40 25
            type hello world
            expected:
            TEA WINDOW SIZE: {40 25}
            -- view:
            > hello world                         ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 25}
            -- view:
            > hello world                         ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/help (0.00s)
        driver.go:474: testdata/help:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/help:1:
             reset
            resize 90 25
            expected:
            TEA WINDOW SIZE: {90 25}
            -- view:
            >                                                                                       ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            
            found:
            TEA WINDOW SIZE: {90 25}
            -- view:
            >                                                                                       ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
    --- FAIL: TestBubbline/hide_cursor (0.00s)
        driver.go:474: testdata/hide_cursor:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/hide_cursor:1:
             reset
            resize 20 10
            expected:
            TEA WINDOW SIZE: {20 10}
            -- view:
            >                 ␤
             …🛇
            
            found:
            TEA WINDOW SIZE: {20 10}
            -- view:
            >                 ␤
             …🛇
    --- FAIL: TestBubbline/hide_show_prompt (0.04s)
        driver.go:474: testdata/hide_show_prompt:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/hide_show_prompt:1:
             reset
            resize 80 25
            type hello
            key ctrl+o
            type world
            output didn't match expected:
            @@ -1,6 +1,6 @@
             TEA WINDOW SIZE: {80 25}
             -- view:
            -> hello                                                                       ␤
            -  world                                                                       ␤
            -M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            +> hello                                                                       ␤
            +  world                                                                       ␤
            +M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
             
    --- FAIL: TestBubbline/history (0.02s)
        driver.go:474: testdata/history:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/history:1:
             reset
            resize 40 25
            type baseline text
            expected:
            TEA WINDOW SIZE: {40 25}
            -- view:
            > baseline text                       ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 25}
            -- view:
            > baseline text                       ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/history_search (0.02s)
        driver.go:474: testdata/history_search:1: applying command "reset" via model updater
        driver.go:474: testdata/history_search:1: applying command "set_history" via model updater
        getline_test.go:32: 
            testdata/history_search:1:
             reset
            resize 40 25
            set_history
            type baseline text
            output didn't match expected:
            @@ -1,9 +1,9 @@
             TEA WINDOW SIZE: {40 25}
             -- view:
            -> baseline text                       ␤
            -M-? toggle key help • C-d erase/stop🛇
            +> baseline text                       ␤
            +M-? toggle key help • C-d erase/stop🛇
             -- history:
             say hello to the world
             peter parker was not spiderman
             this is a big world indeed
             
    --- FAIL: TestBubbline/interrupt (0.04s)
        driver.go:474: testdata/interrupt:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/interrupt:1:
             reset
            resize 80 25
            type hello
            key ctrl+o
            type world
            output didn't match expected:
            @@ -1,6 +1,6 @@
             TEA WINDOW SIZE: {80 25}
             -- view:
            -> hello                                                                       ␤
            -  world                                                                       ␤
            -M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            +> hello                                                                       ␤
            +  world                                                                       ␤
            +M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
             
    --- FAIL: TestBubbline/job_control (0.00s)
        driver.go:474: testdata/job_control:4: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/job_control:4:
             reset
            resize 80 25
            expected:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            
            found:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
    --- FAIL: TestBubbline/max_height (0.00s)
        driver.go:474: testdata/max_height:1: applying command "limit_max_height" via model updater
        driver.go:474: testdata/max_height:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/max_height:1:
             limit_max_height
            reset
            resize 40 25
            expected:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >                                     ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/max_width (0.00s)
        driver.go:474: testdata/max_width:1: applying command "limit_max_width" via model updater
        driver.go:474: testdata/max_width:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/max_width:1:
             limit_max_width
            reset
            resize 40 25
            expected:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >         ␤
            M-? toggle key help • C-d erase/stop🛇
            
            found:
            TEA WINDOW SIZE: {40 25}
            -- view:
            >         ␤
            M-? toggle key help • C-d erase/stop🛇
    --- FAIL: TestBubbline/multiline_paste (0.02s)
        driver.go:474: testdata/multiline_paste:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/multiline_paste:1:
             reset
            resize 40 25
            paste "some text\nmore lines\nagain lines\nfinal line"
            output didn't match expected:
            @@ -1,8 +1,8 @@
             TEA WINDOW SIZE: {40 25}
             -- view:
            -> some text                           ␤
            -  more lines                          ␤
            -  again lines                         ␤
            -  final line                          ␤
            -M-? toggle key help • C-d erase/stop🛇
            +> some text                           ␤
            +  more lines                          ␤
            +  again lines                         ␤
            +  final line                          ␤
            +M-? toggle key help • C-d erase/stop🛇
             
    --- FAIL: TestBubbline/redraw (0.00s)
        driver.go:474: testdata/redraw:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/redraw:1:
             reset
            resize 80 25
            expected:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            
            found:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
    --- FAIL: TestBubbline/reflow (0.00s)
        driver.go:474: testdata/reflow:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/reflow:1:
             reset
            resize 20 25
            expected:
            TEA WINDOW SIZE: {20 25}
            -- view:
            >                 ␤
             …🛇
            
            found:
            TEA WINDOW SIZE: {20 25}
            -- view:
            >                 ␤
             …🛇
    --- FAIL: TestBubbline/simple (0.00s)
        driver.go:474: testdata/simple:1: applying command "reset" via model updater
        getline_test.go:32: 
            testdata/simple:1:
             reset
            resize 80 25
            expected:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
            
            found:
            TEA WINDOW SIZE: {80 25}
            -- view:
            >                                                                             ␤
            M-? toggle key help • C-d erase/stop • C-c clear/cancel • M-. hide/show prompt🛇
FAIL
exit status 1
FAIL	github.com/knz/bubbline/editline	0.303s
```

Reply:
Looks like it works when running:

```
$ /usr/local/go/bin/go test -timeout 30s -run ^TestBubbline$ github.com/knz/bubbline/editline
ok      github.com/knz/bubbline/editline        1.750s
```